packages/core/src/hydration/utils.ts

```diff
@@ -68,6 +68,14 @@ export const NGH_DEFER_BLOCKS_KEY: StateKey<{[key: string]: SerializedDeferBlock
   [key: string]: SerializedDeferBlock;
 }>(TRANSFER_STATE_DEFER_BLOCKS_INFO);
 
+/**
+ * Set of transfer state keys that are used internally by the framework
+ */
+export const internalHydrationTransferStateKeys = [
+  TRANSFER_STATE_TOKEN_ID,
+  TRANSFER_STATE_DEFER_BLOCKS_INFO,
+];
+
```

Comment:
I'd like to propose a couple of changes:

- Create a function (instead of an array) as mentioned below
- Use the function to filter out such keys from the transfer state (in the logic inside of `getTransferState`)
- For tests, we can use regular hydration specs as a base, just add a few more keys to the transfer state and after that assess whether `getTransferState` returns only public/application keys

```suggestion
/**
 * Checks whether a given key is used by the framework for transferring hydration data.
 */
export function isInternalHydrationTransferStateKey(key: string): boolean {
  return key === TRANSFER_STATE_TOKEN_ID || key === TRANSFER_STATE_DEFER_BLOCKS_INFO;
}
```

Reply:
I've extracted the core part of this PR into a separate PR #62722